src/interfaces/interfaces.ts

```diff
@@ -61,8 +61,8 @@ namespace interfaces {
         dynamicValue: DynamicValue<TActivated> | null;
         scope: BindingScope;
         type: BindingType;
-        implementationType: Newable<TActivated> | null;
-        factory: FactoryCreator<TActivated> | null;
+        implementationType: Newable<TActivated> | TActivated | null;
+        factory: FactoryCreator<any> | null;
```

Comment:
i'm just curios. can we not use `unknown` here ?

Reply:
I guess it would force adding some castings